chain/chain/src/chain.rs

```diff
@@ -506,6 +503,10 @@ pub struct Chain {
     snapshot_callbacks: Option<SnapshotCallbacks>,
 
     pub(crate) state_split_config: near_chain_configs::StateSplitConfig,
+
+    /// If the node is not a chunk producer within that many blocks, then route
+    /// to upcoming chunk producers.
+    tx_routing_height_horizon: Option<BlockHeightDelta>,
```

Comment:
I slightly prefer `BlockHeightDelta` because we only unwrap it and we have a default function `default_tx_routing_height_horizon` anyway.

Or use explicit match in `find_chunk_producer_for_forwarding` and return an Error if `tx_routing_height_horizon` is None. unwrap just looks less safe.